sympy/polys/matrices/tests/test_domainmatrix.py

```diff
@@ -440,3 +440,19 @@ def test_DomainMatrix_zeros():
     assert A.rep == SDM.zeros((1, 2), QQ)
     assert A.shape == (1, 2)
     assert A.domain == QQ
+
+def test_DomainMatrix_getitem():
+    dm = DomainMatrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]], (3, 3), ZZ)
+    assert dm[::-1, :] == dm
+    assert dm[1:,:-3] == DomainMatrix([[4], [7]], (2, 1), ZZ)
+    assert dm[2,:-3] == DomainMatrix([[7]], (1, 1), ZZ)
+    assert dm[:-2,:-3]==DomainMatrix([[1], [4]], (2, 1), ZZ)
+    assert dm[:-2,0:1] == DomainMatrix([[1, 2], [4, 5]], (2, 2), ZZ)
+    #assert dm[1, 2] == DomainScalar(6, dm.domain)
```

Comment:
This raises assertion error, but not able to understand why?
```
In [1]: from sympy.polys.matrices.domainmatrix import DomainMatrix, DomainScalar

In [2]: dm = DomainMatrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]], (3, 3), ZZ)

In [3]: dm[1, 2] == DomainScalar(6, dm.domain)
Out[3]: False

In [4]: dm[1, 2]
Out[4]: 6

In [5]: type(dm[1, 2])
Out[5]: sympy.polys.matrices.domainmatrix.DomainScalar

In [6]: DomainScalar(6, dm.domain)
Out[6]: 6

In [7]: type(DomainScalar(6, dm.domain))
Out[7]: sympy.polys.matrices.domainmatrix.DomainScalar

In [8]: dm[1, 2].domain
Out[8]: ZZ

In [9]: DomainScalar(6, dm.domain).domain
Out[9]: ZZ

```

Reply:
DomainScalar needs an `__eq__` method.

Reply:
Sure, thanks.